apps/hash-graph/lib/graph/src/store/validation.rs

```diff
@@ -52,33 +137,61 @@ where
                 .change_context(QueryError)?;
         }
 
-        self.store
+        Ok(())
+    }
+}
+
+impl<S, A> OntologyTypeProvider<DataType> for StoreProvider<'_, S, A>
+where
+    S: Read<DataTypeWithMetadata, Record = DataTypeWithMetadata>,
+    A: AuthorizationApi + Sync,
+{
+    #[expect(refining_impl_trait)]
+    async fn provide_type(
+        &self,
+        type_id: &VersionedUrl,
+    ) -> Result<Arc<DataType>, Report<QueryError>> {
+        let data_type_id = DataTypeId::from_url(type_id);
+
+        if let Some(cached) = self.cache.data_types.get(&data_type_id).await {
+            return cached;
+        }
+
+        if let Err(error) = self.authorize_data_type(data_type_id).await {
+            self.cache.data_types.deny(data_type_id).await;
+            return Err(error);
+        }
+
+        let type_ = self
```

Comment:
Let's avoid the weird underscore
```suggestion
        let schema = self
```

Reply:
[204910b](https://github.com/hashintel/hash/pull/3633/commits/204910b84646c5601c7c81c274faba86df980b30)